src/ValueHandler/ProductOptionValueHandler.php

```diff
@@ -161,16 +210,137 @@ public function handle($productVariant, string $optionCode, array $akeneoValue):
+        //            }
+        //        )->first();
+        if ($productOption === null) {
+            throw new RuntimeException(
+                sprintf(
+                    'Cannot import Akeneo product "%s", the option "%s" is not set on the parent product "%s".',
+                    (string) $productVariant->getCode(),
+                    $optionCode,
+                    (string) $product->getCode()
+                )
+            );
+        }
+
+        return $productOption;
+    }
+
+    private function getOrCreateProductOptionValue(string $optionValueCode, ProductOptionInterface $productOption): ProductOptionValueInterface
+    {
+        /** @var ProductOptionValueInterface|null $optionValue */
+        $optionValue = $this->productOptionValueRepository->findOneBy(['code' => $optionValueCode]);
+        if (!$optionValue instanceof ProductOptionValueInterface) {
+            /** @var ProductOptionValueInterface $optionValue */
+            $optionValue = $this->productOptionValueFactory->createNew();
+            $optionValue->setCode($optionValueCode);
+            $optionValue->setOption($productOption);
+            $productOption->addValue($optionValue);
+        }
+
+        return $optionValue;
+    }
+
+    private function getLocaleCodes(ProductInterface $product): array
+    {
+        $locales = [];
+        if ($this->translationLocaleProvider !== null) {
+            $locales = $this->translationLocaleProvider->getDefinedLocalesCodes();
+        } else {
+            foreach ($product->getTranslations() as $translation) {
+                $locale = $translation->getLocale();
+                if ($locale === null) {
+                    continue;
+                }
+                $locales[] = $locale;
+            }
+        }
+
+        return $locales;
+    }
+
+    private function addOptionValueTranslation(
+        ProductOptionValueInterface $optionValue,
+        string $localeCode,
+        string $label
+    ): ProductOptionValueInterface {
+        $optionValueTranslation = $optionValue->getTranslation($localeCode);
+        if ($optionValueTranslation->getLocale() !== $localeCode) {
+            /** @var ProductOptionValueTranslationInterface $optionValueTranslation */
+            $optionValueTranslation = $this->productOptionValueTranslationFactory->createNew();
+            $optionValueTranslation->setLocale($localeCode);
+        }
+        $optionValueTranslation->setValue($label);
+        if (!$optionValue->hasTranslation($optionValueTranslation)) {
+            $optionValue->addTranslation($optionValueTranslation);
+        }
+
+        return $optionValue;
+    }
+
+    private function createOptionValueCode(string ...$pieces): string
+    {
+        $slugifiedPieces = array_map(static function (string $word): ?string {
+            return preg_replace('/[^a-z0-9\-_]+/i', '', $word);
+        }, $pieces);
+
+        return implode('_', $slugifiedPieces);
```

Comment:
I'm afraid we can't because option codes can only be letters, numbers, dashes and underscores. Anyway I think this is a very unlikely case that we can safely ignore.

Reply:
Or... maybe we should simplify it: considering that this issue was raised by the introduction of metric attributes, and that the only characters that we need to exclude are characters that appear in numeric values, that's to say dots and maybe  commas, wouldn't it be better to make a simple `str_replace` to remove only dots and commas? What do you think @LucaGallinari and @azambon ?

Reply:
We should also be sure to not introduce a BC break by changing the logic the production option value code is generated.

Reply:
The only BC could be on the "slugification" of the option value code for select attribute, cause boolean and metric values were not supported.

For select attributes, before we did: 
```
$fullValueCode = $optionCode . '_' . $partialValueCode;
```
where $partialValueCode is the code of the value. Given that $optionCode and $partialValueCode are two codes, they could not (and actually cannot) contains dots and points.
Now we do:
```
$slugifiedPieces = array_map(static function (string $word): string {
    return str_replace(['.', ','], '', $word);
}, $pieces);

return implode('_', $slugifiedPieces);
```
where $pieces "contains" $optionCode and $partialValueCode, and the new str_replace is idempotent for select attributes cause, as said before, the two codes cannot contain dots and points. Thus, the result is the same before and now (this is "softly" asserted in some phpspec tests that checks the generated code for select attribute, [here](https://github.com/webgriffe/SyliusAkeneoPlugin/pull/132/files#diff-276ac0d5755f3906c70247804ffe9bf94aaf3879f107067bda3bca5cb8d9acc7R273) and [here](https://github.com/webgriffe/SyliusAkeneoPlugin/pull/132/files#diff-276ac0d5755f3906c70247804ffe9bf94aaf3879f107067bda3bca5cb8d9acc7R346))